### PR TITLE
Fixed children removal in the attachShadow method

### PR DESCRIPTION
**Pull request by adevnadia**

### CODE REVIEW

File: test/unit.js [hunk 1]
```diff
@@ -8,13 +8,31 @@ import './unit/slot/distribution';
...
 import './unit/webcomponents/initialisation';
+import canPatchNativeAccessors from '../src/util/can-patch-native-accessors';
 
 describe('skatejs-named-slots', function () {
   it('version', function () {
     expect(version).to.be.a('string');
   });
 });
 
+if (canPatchNativeAccessors) {
```
Comment (treshugart) [inline test/unit.js]: A better place to put this test might be in the `test/unit/shadow/polyfill.js` file as it seems to be specifically testing what happens to the original nodes when `attachShadow()` is called.

File: test/unit.js [hunk 2]
```diff
@@ -8,13 +8,31 @@ import './unit/slot/distribution';
...
+    it('proper node removal', function () {
+
+      const host = create('div');
+      host.appendChild(create('div'));
+      host.appendChild(create('div'));
+      host.appendChild(create('div'));
+      host.appendChild(create('div'));
+      host.attachShadow({ mode: 'open' });
+      
+      expect(host.__childNodes.length).to.equal(1);
```
Comment (treshugart) [inline test/unit.js]: It would be worth asserting whether or not the only child node is actually the `shadowRoot`.

File: src/index.js [hunk 3]
```diff
@@ -487,7 +487,12 @@ const members = {
       // appear to be child nodes. This is how light DOM works; they're still
       // child nodes but not in the composed DOM yet as there won't be any
       // slots for them to go into.
-      eachChildNode(this, node => this.__removeChild(node));
+      const that = this;
```
Comment (treshugart) [inline src/index.js]: Not required as you can use `this` in the `for` loop.

**Comment by jpnelson** [comment]
lgtm

File: test/unit/shadow/polyfill.js [hunk 4]
```diff
@@ -38,6 +39,21 @@ describe('shadow/polyfill', function () {
...
+    it('proper node removal', function () {
+      const host = create('div');
+      host.appendChild(create('div'));
+      host.appendChild(create('div'));
+      host.appendChild(create('div'));
+      host.appendChild(create('div'));
+      host.attachShadow({ mode: 'open' });
+
+      expect(host.__childNodes.length).to.equal(1);
+      expect(host.__childNodes[0].tagName).to.equal('_SHADOW_ROOT_');
```
Comment (treshugart) [inline test/unit/shadow/polyfill.js:52]: A more robust check would be to do:

``` js
expect(host.__childNodes[0]).to.equal(host.shadowRoot);
```

Or, you could use the return value of `attachShadow()` as the `shadowRoot`.

The only reason I pose this is if the name of the shadow root tag name changes then the test breaks.

**Comment by treshugart** [comment]
lgtm

**Pull request merged**